src/libtrx/game/level/common.c

```diff
@@ -0,0 +1,80 @@
+#include "game/level/common.h"
+
+#include "game/gamebuf.h"
+#include "game/inject.h"
+#include "game/rooms.h"
+
+#include <assert.h>
+
+void Level_ReadRoomMesh(const int32_t room_num, VFILE *const file)
+{
+#if TR_VERSION == 1
+    ROOM *const room = Room_Get(room_num);
+    const INJECTION_MESH_META inj_data = Inject_GetRoomMeshMeta(room_num);
+
+    const uint32_t mesh_length = VFile_ReadU32(file);
+    size_t start_pos = VFile_GetPos(file);
+
+    {
+        room->mesh.num_vertices = VFile_ReadS16(file);
+        const int32_t alloc_count =
+            room->mesh.num_vertices + inj_data.num_vertices;
+        room->mesh.vertices =
+            GameBuf_Alloc(sizeof(ROOM_VERTEX) * alloc_count, GBUF_ROOM_MESH);
+        for (int32_t i = 0; i < room->mesh.num_vertices; i++) {
+            ROOM_VERTEX *const vertex = &room->mesh.vertices[i];
+            vertex->pos.x = VFile_ReadS16(file);
+            vertex->pos.y = VFile_ReadS16(file);
+            vertex->pos.z = VFile_ReadS16(file);
+            vertex->shade = VFile_ReadU16(file);
+            vertex->flags = 0;
+        }
+    }
```

Comment:
Are these outer parentheses just used so you can reuse variable names by reducing the scope of each block?

Reply:
Yes, pretty much. Short of separating the four property reads into functions, I think this is OK to group the logic together while still making scope clear.